test/dotcom/utils/service_date_time_test.exs

```diff
@@ -0,0 +1,221 @@
+defmodule Dotcom.Utils.ServiceDateTimeTest do
+  use ExUnit.Case
+  use ExUnitProperties
+
+  import Mox
+  import Dotcom.Utils.ServiceDateTime
+  import Test.Support.Generators.DateTime
+  import Test.Support.Generators.ServiceDateTime
+
+  setup _ do
+    stub_with(Dotcom.Utils.DateTime.Mock, Dotcom.Utils.DateTime)
+
+    :ok
+  end
+
+  describe "service_rollover_time/0" do
+    test "returns a time for the service rollover time" do
+      # Exercise/Verify
+      assert %Time{} = service_rollover_time()
+    end
+  end
+
+  describe "service_date/1" do
+    property "returns 'today' when the date_time is between the start of service and midnight" do
+      # Setup
+      check all(date_time <- date_time_generator(:before_midnight)) do
+        beginning_of_service_day = beginning_of_service_day(date_time)
+        end_of_day = Timex.end_of_day(date_time)
+
+        date_time_generator =
+          time_range_date_time_generator({beginning_of_service_day, end_of_day})
+
+        check all(service_date_time <- date_time_generator) do
+          # Exercise
+          date = Timex.to_date(date_time)
+          service_date = service_date(service_date_time)
+
+          # Verify
+          assert Timex.equal?(date, service_date, :day)
+        end
+      end
+    end
+
+    property "returns 'yesterday' when the date_time is between midnight and the end of service" do
+      # Setup
+      check all(date_time <- date_time_generator(:after_midnight)) do
+        beginning_of_day = Timex.end_of_day(date_time) |> Timex.shift(microseconds: 1)
+        end_of_service_day = end_of_service_day(date_time)
+
+        date_time_generator =
+          time_range_date_time_generator({beginning_of_day, end_of_service_day})
+
+        check all(service_date_time <- date_time_generator) do
+          yesterday = service_date_time |> Timex.shift(days: -1)
+
+          # Exercise
+          date = Timex.to_date(yesterday)
+          service_date = service_date(date_time)
+
+          # Verify
+          assert Timex.equal?(date, service_date, :day)
+        end
+      end
+    end
+  end
+
+  describe "service_range/1" do
+    test "returns :before_today for before_today" do
+      # Setup
+      today = beginning_of_service_day()
+      before_today = Timex.shift(today, microseconds: -1)
+
+      # Exercise/Verify
+      assert service_range(before_today) == :before_today
+    end
+
+    test "returns :today for today" do
+      # Setup
+      today = service_range_day() |> random_time_range_date_time()
+
+      # Exercise/Verify
+      assert service_range(today) == :today
+    end
+
+    test "returns :today if today is the last day of the service week" do
+      # Setup
+      expect(Dotcom.Utils.DateTime.Mock, :now, 2, fn ->
```

Comment:
Suggestion: Instead of mocking `now` twice in this test, pass `Dotcom.Utils.DateTime.now() |> Timex.beginning_of_week() |> beginning_of_service_day()` into `service_range_later_this_week()` as an argument. 

I think that in order for that to work fully, we'd need to replace the `expect` calls with `stub` in this test.